torch/utils/checkpoint.py

```diff
@@ -206,18 +212,30 @@ def checkpoint(function, *args, use_reentrant: bool = True, **kwargs):
             first input as ``activation`` and the second input as ``hidden``
         preserve_rng_state(bool, optional, default=True):  Omit stashing and restoring
             the RNG state during each checkpoint.
-        use_reentrant(bool, optional, default=True): Use checkpointing
+        use_reentrant(bool default=True): Use checkpointing
             implementation that requires re-entrant autograd.
             If ``use_reentrant=False`` is specified, ``checkpoint`` will use an
             implementation that does not require re-entrant autograd. This
             allows ``checkpoint`` to support additional functionality, such as
-            working as expected with ``torch.autograd.grad``. Note that future
-            versions of PyTorch will default to ``use_reentrant=False``.
+            working as expected with ``torch.autograd.grad``. Default is
+            currently ``True`` for backwards compatibility, but note that
+            omitting the argument is deprecated and it should be explicitly
+            specified. Future versions of PyTorch will default to
+            ``use_reentrant=False``.
         args: tuple containing inputs to the :attr:`function`
 
     Returns:
         Output of running :attr:`function` on :attr:`*args`
     """
+
+    if use_reentrant:
```

Comment:
This is not good as users that want to use the reentrant version will not be able to silence the warning.
You should make the default `None` and warn when the value is None here and set it to True.